lib/services/sku-pack-service.js

```diff
@@ -95,5 +132,96 @@ function skuPackServiceFactory(
+
+    SkuPackService.prototype.installPack = function(fromRoot, skuid) {
+        var self = this;
+        return fs.readFileAsync(fromRoot + '/config.json')
+            .then(function(contents) {
+                return self.validatePack(contents, fromRoot)
+                    .then(function() {
+                        if(skuid === undefined)  {
+                            var conf = JSON.parse(contents);
+                            if(!conf.hasOwnProperty('rules') || !conf.hasOwnProperty('name')) {
+                                throw new Error('rules or name is missing');
+                            }
+                            return waterline.skus.create({name: conf.name, rules: conf.rules})
+                                .then(function(sku) {
+                                    return sku.id;
+                                });
+                        }
+                        return skuid;
+                    })
+                    .then(function(skuID) {
+                        return fs.statAsync(self.confRoot + '/' + skuID).then(function(stat) {
+                                if(stat.isDirectory())  {
+                                    return self.deletePack(skuID);
+                                }
+                                return skuID;
+                            })
+                            .catch(function() {
+                                return skuID;
+                            });
+                    })
+                    .then(function(skuID) {
+                        return [skuID, fs.readdirAsync(fromRoot)];
+                    })
+                    .spread(function(sku, entries) {
+                        var dst = self.confRoot + '/' + sku;                        
+                        return fs.mkdirAsync(dst)
+                            .then(function() {
+                                return Promise.all(_.map(entries, function(entry) {
+                                    var src = fromRoot + '/' + entry;
+                                    if( entry === 'config.json' ) {
+                                        return fs.renameAsync(src, dst + '.json');
+                                    } else {
+                                        return fs.renameAsync(src, dst + '/' + entry);
+                                    }
+                                }));
+                            })
+                            .then(function() {
+                                return [dst + '.json', contents];
+                            });
+                    });
+                });
+    };
+
+    SkuPackService.prototype.deletePack = function(skuid) {
+        var self = this;
+        return fs.readFileAsync(self.confRoot + '/' + skuid + '.json')
+            .then(function(contents) {
+                return self.unregisterPack(contents, skuid);
+            })
+            .then(function() {
+                return rimrafAsync(self.confRoot + '/' + skuid);
```

Comment:
Just to be extra careful, do you think it's worth asserting that `self.confRoot` is not null or an empty string?